src/mango/src/mango_cursor_view.erl

```diff
@@ -209,22 +213,66 @@ choose_best_index(_DbName, IndexRanges) ->
     {SelectedIndex, SelectedIndexRanges}.
 
 
+baseViewRow(Row) ->
+    #view_row{
+        id = couch_util:get_value(id, Row),
+        key = couch_util:get_value(key, Row),
+        doc = couch_util:get_value(doc, Row)
+    }.
+
+
+view_cb({meta, Meta}, Acc) ->
+    % Map function starting
+    put(mango_docs_examined, 0),
+    ok = rexi:stream2({meta, Meta}),
+    {ok, Acc};
+view_cb({row, Row}, #mrargs{extra = Options} = Acc) ->
+    ViewRow = baseViewRow(Row),
+    case couch_util:get_value(doc, Row) of
+        undefined ->
+            ViewRow2 = ViewRow#view_row{
+                value = couch_util:get_value(value, Row)
+            },
+            ok = rexi:stream2(ViewRow2),
+            put(mango_docs_examined, 0);
+        Doc ->
+            Selector = couch_util:get_value(selector, Options),
+            case mango_selector:match(Selector, Doc) of
+                true ->
+                    ViewRow2 = ViewRow#view_row{
+                        value = get(mango_docs_examined) + 1
+                    },
+                    ok = rexi:stream2(ViewRow2),
+                    put(mango_docs_examined, 0);
+                false ->
+                    put(mango_docs_examined, get(mango_docs_examined) + 1)
```

Comment:
Something to think about for the future will be how to tunnel a message through from here occasionally if we happen upon a highly restrictive filter. This doesn't change anything from the user POV currently as we could have the same effect from filtering at the collector level.